src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -426,13 +426,68 @@ internal void AddInterfaceImpl(Type iface)
+                var propertyMap = new Dictionary<MethodInfo, PropertyAccessorInfo>();
+                var eventMap = new Dictionary<MethodInfo, EventAccessorInfo>();
+                foreach (var pi in iface.GetRuntimeProperties())
+                {
+                    var ai = new PropertyAccessorInfo(pi.GetMethod, pi.SetMethod);
+                    if (pi.GetMethod != null) propertyMap[pi.GetMethod] = ai;
+                    if (pi.SetMethod != null) propertyMap[pi.SetMethod] = ai;
+                }
+
+                foreach (var ei in iface.GetRuntimeEvents())
+                {
+                    var ai = new EventAccessorInfo(ei.AddMethod, ei.RemoveMethod);
+                    if (ei.AddMethod != null) eventMap[ei.AddMethod] = ai;
+                    if (ei.RemoveMethod != null) eventMap[ei.RemoveMethod] = ai;
+                }
+
                 foreach (MethodInfo mi in iface.GetRuntimeMethods())
                 {
-                    AddMethodImpl(mi);
+                    var mdb = AddMethodImpl(mi);
+                    PropertyAccessorInfo associatedProperty;
+                    if (propertyMap.TryGetValue(mi, out associatedProperty))
+                    {
+                        if (associatedProperty.InterfaceGetMethod == mi)
+                            associatedProperty.GetMethodBuilder = mdb;
+                        else
+                            associatedProperty.SetMethodBuilder = mdb;
+                    }
+
+                    EventAccessorInfo associatedEvent;
+                    if (eventMap.TryGetValue(mi, out associatedEvent))
+                    {
+                        if (associatedEvent.InterfaceAddMethod == mi)
+                            associatedEvent.AddMethodBuilder = mdb;
+                        else
+                            associatedEvent.RemoveMethodBuilder = mdb;
+                    }
+                }
+
+                foreach (var pi in iface.GetRuntimeProperties())
+                {
+                    var ai = propertyMap[pi.GetMethod ?? pi.SetMethod];
+                    var pb = _tb.DefineProperty(pi.Name, pi.Attributes, pi.PropertyType, pi.GetIndexParameters().Select(p => p.ParameterType).ToArray());
+                    if (ai.GetMethodBuilder != null)
+                        pb.SetGetMethod(ai.GetMethodBuilder);
+                    if (ai.SetMethodBuilder != null)
+                        pb.SetSetMethod(ai.SetMethodBuilder);
+                }
+
+                foreach (var ei in iface.GetRuntimeEvents())
+                {
+                    var ai = eventMap[ei.AddMethod ?? ei.RemoveMethod];
+                    var eb = _tb.DefineEvent(ei.Name, ei.Attributes, ei.EventHandlerType);
+                    if (ai.AddMethodBuilder != null)
+                        eb.SetAddOnMethod(ai.AddMethodBuilder);
+                    if (ai.RemoveMethodBuilder != null)
+                        eb.SetRemoveOnMethod(ai.RemoveMethodBuilder);
```

Comment:
I can implement it easily, but it is hard to test because C# compiler does not emit "raising event" method as long as I know. So I need to emit raise method by IL generation. It looks overkill...